spring-kafka/src/main/java/org/springframework/kafka/support/serializer/JsonDeserializer.java

```diff
@@ -110,11 +114,28 @@ public Jackson2JavaTypeMapper getTypeMapper() {
 	public void setTypeMapper(Jackson2JavaTypeMapper typeMapper) {
 		Assert.notNull(typeMapper, "'typeMapper' cannot be null");
 		this.typeMapper = typeMapper;
+		this.typeMapperExplicitlySet = true;
+	}
+
+	/**
+	 * Configure the JsonDeserializer to use key type headers.
+	 * @param isKey Use key type headers if true
+	 */
+	public void setUseTypeMapperForKey(boolean isKey) {
```

Comment:
Do you think this setting should be moved to ```AbstractJavaTypeMapper```? I'm repeating myself in ```JsonSerializer``` & ```JsonSerializer```

Reply:
Right, sounds good.
Delegating is the best choice here do not repeate 😄 

Although maybe with a slightly different name - `setUseForKey(boolean)`. Since the class is `TypeMapper` per se already.

Reply:
Need `@since 2.1.3` in JavaDocs, too.